### PR TITLE
fix: preserve list meanings

### PR DESCRIPTION
**Pull request by daniel-sanche**
Currently, the library only keeps track of the `meaning` field of sub-elements of array_values it sees. We also need it to keep track of the `meaning` of the array itself

This PR addresses this by returning a tuple of `(root_meaning, sub_meaning_list)` instead of just the sub_meaning_list in `_get_meaning()`, and using the tuple to populate the fields properly in `_set_pb_meaning_from_entity()`

### CODE REVIEW

**Review by cindy-peng — APPROVED** [review]

File: google/cloud/datastore/helpers.py [hunk 1]
```diff
@@ -43,36 +42,29 @@ def _get_meaning(value_pb, is_list=False):
...
-        else:  # We know len(value_pb.array_value.values) > 0.
-            # If the meaning is not unique, just return all of them.
-            return all_meanings
-
-    elif value_pb.meaning:  # Simple field (int32).
-        return value_pb.meaning
-
-    return None
+        sub_meanings = [sub_value_pb.meaning or None for sub_value_pb in values]
+        if not any(meaning is not None for meaning in sub_meanings):
```
Comment (NickChittle) [inline google/cloud/datastore/helpers.py:59]: I'm curious, why wouldn't we just return a tuple of (None, []), instead of transforming the return type here to just a single None?

Comment (daniel-sanche) [reply google/cloud/datastore/helpers.py:59]: In case it's not clear: the reason for this transformation is to flatten giant lists of `[None, None, None, ...]`, so we don't have to iterate over the whole thing again later

You're right that we could return an empty list instead of None here to signal that there are no meanings present. But it's a trade-off. That would buy us type consistency, but then we'd lose the property that the sub-meaning list is always be the same size as the original array. We'd need to remember to do size comparisons before iterating, so we don't really win anything from that change. And size mismatches wouldn't be flagged by type checkers like None checks, so those bugs would be harder to notice.

TL;DR: This is a special case in our processing logic, and it's usually better to use None to signal special cases, instead of returning lists with unpredictable sizes
